### DISCUSSION TITLE
Custom fetch function in HTTP link doesn't seem to work

### DISCUSSION THREAD

**Discussion by gino** [trpc/trpc#2752] [Q&A] [answered]
Hi there,

I recently updated my packages to the new v10 version that uses HTTP links and tried to convert the following old code:

```ts
return trpc.createClient({
  url: "http://localhost:3001/trpc",
  transformer: superjson,
  fetch(url, options) {
    return fetch(url, {
      ...options,
      credentials: "include",
    });
  },
});
```

into:

```ts
return trpc.createClient({
  links: [
    httpBatchLink({
      url: "http://localhost:3001/trpc",
      fetch(url, options) {
        return fetch("an url that should fail", {
          ...options,
          credentials: "include",
        });
      },
    }),
  ],
  transformer: superjson,
});
```

I realized that my whole authentication flow broke after making those changes and I tested if the custom fetch function (that should also include the credentials property for cookie purposes) was being used at all. And it looks like it isn't. The requests are still being made to the correct URL, even though it should fail in the example above (I've set an incorrect URL to test it out):

<img width="312" alt="image" src="https://user-images.githubusercontent.com/23482089/191035863-083bf031-b7b4-40ba-b5c3-b6426de68bcb.png">

If that's the case, this could explain why my cookies are no longer being set cause the `credentials` property is not being set at all.

I think I converted correctly to the new HTTP links unless I'm missing something, not quite sure.

Hope someone is able to help me out here, thanks a lot.

**Comment by gino** [comment]
**Edit:** I just tried to reproduce this in a different React project and the fetch function does seem to work there.. not quite sure what's the difference between my project where it doesn't work but will figure it out. Will update on this later.

**Reply by KATT** [reply]
Thanks for the update! Let us know how it works

**Comment by gino** [comment, marked as answer]
Apparently, there was an issue with my `node_modules` folder. Couldn't figure out what was wrong since I had no type errors at all. Quite a strange issue but solved it by manually removing the `node_modules` folder and running `pnpm install` again. 

I guess something went wrong when I updated all the packages. Sorry for the inconvenience.

**Reply by KATT** [reply]
Classic 😅
